### PR TITLE
feat: support docker

### PR DESCRIPTION
**Pull request by muhammad-asn**
Add support for Docker

### CODE REVIEW

**Comment by muhammad-asn** [comment]
@technikhil314 Please kindly check this PR. It will enable the offline-diff-viewer to run as a container-based application.

**Comment by technikhil314** [comment]
Just trying to understand the use case here. Isnt just going to site and seeing the diff there more easy and convenient? @muhammad-asn

**Comment by muhammad-asn** [comment]
> Just trying to understand the use case here. Isnt just going to site and seeing the diff there more easy and convenient? @muhammad-asn

I think It will help if someone want to deploy (self-hosted) using containerization.

File: nuxt.config.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const canonicalLinks = domainAliases.map((x) => ({ rel: 'canonical', href: x }))
 const DESCRIPTION =
   'A privacy first diff viewer that is secure, easy, simple and for any text type'
 export default {
+  target: 'static',
```
[the remark this answers is not in the document]
Comment (muhammad-asn) [reply nuxt.config.js]: I don't have access to vercel. So i think it's expected the build was failed

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM node:16 AS build-stage
+
+WORKDIR /app
+
+COPY package*.json ./
+
+RUN npm install
```
[the remark this answers is not in the document]
Comment (muhammad-asn) [reply Dockerfile]: done

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM node:16 AS build-stage
```
[the remark this answers is not in the document]
Comment (muhammad-asn) [reply Dockerfile]: already updated

File: nuxt.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (technikhil314) [reply nuxt.config.js]: oh got it. just authorized the workflow. Thanks.

File: README.md [hunk 4]
```diff
@@ -76,4 +76,7 @@ $ npm run start
 
 # generate static project
 $ npm run generate
+
+# serve with docker
+$ docker compose up -d --build
 ```
```
[the remark this answers is not in the document]
Comment (muhammad-asn) [reply README.md:82]: I see alright will update it

[the remark this answers is not in the document]
Comment (muhammad-asn) [reply README.md:88]: done

[the remark this answers is not in the document]
Comment (technikhil314) [reply README.md:88]: Great thanks a lot. Will try it myself and merge over next few days.

**Review by technikhil314 — APPROVED** [review]

**Pull request merged**